caput/memh5.py

```diff
@@ -2514,7 +2514,7 @@ def _prepare_dataset(dset):
                 return dset.dtype, dset.shape, None
 
         # Extract the data for the selection
-        data = entry[selection]
```

Comment:
huh, so I guess this just worked because it was being called in such a way that `entry` was correctly defined. How did you spot this?

Reply:
Just by reading through the code nearby. But yeah, you're write that it's only the fact the variable if function scoped that it worked.